### PR TITLE
feat(MCDU): automatic add sid and star from simbrief

### PR DESCRIPTION
**Pull request by igor8518**
Fixes #[issue_no]

## Summary of Changes
This PR adds the functionality of automatically adding SID/STAR from SimBrief. Also clears the flightplan from duplicates of waypoints and Discontinuity.

TO DO

- [x] From SimBrief
    - [x] Add SID functionality 
    - [x] Add STAR functionality 
- [x] Misc
    - [x] Clear flightplan from discontinuity markers
    - [x] Clear duplicates of waypoints
    - [x] Add option to EFB realism page
    - Working out the logic
    - Code refactoring

**X** Generated (Next generation)
   - **X** Add SID functionality 
   - **X** Add STAR functionality 

## Additional context

Discord username: igor8518#8492

## Testing instructions
When preparing a flight on the Init page from mcdu, click request. The full route with SID and STAR should appear on the FlightPlan page.
**Testing is required at various airports. There may be problems. Please specify the route and AIRAC cycle.**

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by FreshDave29** [comment]
Is the deletion of discos an option you can choose? This sounds like a special behaviour and is not replicated from the original aircraft (which allows route insertion via company message).
The route discos are there for a reason, mainly to remind of any clearance limit. Although, many company procedures allow to delete them while not yet departed.

**Comment by igor8518** [comment]
> Является ли удаление дискотек вариантом, который вы можете выбрать? Это звучит как особое поведение и не копируется с оригинального самолета (что позволяет вводить маршрут через сообщение компании). Дискотеки на маршруте существуют не просто так, главным образом для того, чтобы напоминать о любом ограничении доступа. Хотя многие процедуры компании позволяют удалять их, пока они еще не ушли.

I suggest making the last two items configurable via EFB.

**Comment by Benjozork** [comment]
It is important to remember that discontinuity logic is not perfect right now and will be vastly improved in the next flight plan manager version. So this code should be temporary if it is included.

**Comment by tracernz** [comment]
The reason we do not uplink SID/STAR/approach is because all of the pilots who advised us indicated that their route uplinks do not include those.

**Comment by igor8518** [comment]
Perhaps I will make this functionality optional.

File: src/instruments/src/EFB/Settings/Settings.tsx [hunk 1]
```diff
@@ -467,6 +479,36 @@ const RealismPage = () => {
                             ))}
                         </SelectGroup>
                     </div>
+
+                    <div className="py-4 flex flex-row justify-between items-center">
+                        <span className="text-lg text-gray-300 mr-1">Automatic delete discontimuity on flightplan SimBrief init</span>
```
Comment (beheh) [inline src/instruments/src/EFB/Settings/Settings.tsx:484]:
```suggestion
                        <span className="text-lg text-gray-300 mr-1">Automatically delete discontinuity on SimBrief import</span>
```

File: src/instruments/src/EFB/Settings/Settings.tsx [hunk 2]
```diff
@@ -467,6 +479,36 @@ const RealismPage = () => {
...
+                                    selected={autoDeleteDiscontinuity === button.setting}
+                                >
+                                    {button.name}
+                                </SelectItem>
+                            ))}
+                        </SelectGroup>
+                    </div>
+
+                    <div className="py-4 flex flex-row justify-between items-center">
+                        <span className="text-lg text-gray-300 mr-1">Automatic load or generated SID/STAR</span>
```
Comment (beheh) [inline src/instruments/src/EFB/Settings/Settings.tsx:499]:
```suggestion
                        <span className="text-lg text-gray-300 mr-1">Automatically import SIDs and STARs from SimBrief</span>
```

**Review by beheh — COMMENTED** [review]
Just two typos/rewording suggestions for now.

**Comment by beheh** [comment]
On a high-level as a user I'm not sure why I should check "Automatically delete discontinuities on SimBrief import". When would a SimBrief flight plan contain a discontinuity?

**Comment by igor8518** [comment]
In fact, this PR is part of another project where the co-pilot's work is simulated, up to manual control of the aircraft on the ground and in the air (https://m.youtube.com/watch?v=0ny_zmtxz4k&t=4168s). I can do import and cleanup management only through L:variables, so this PR will not be noticeable to the user.

**Comment by davidwalschots** [comment]
This PR is not a draft, but has TODO items that remain unchecked in its description. Are you expecting a code review at this point in time?

**Comment by igor8518** [comment]
I'm thinking not to do autogeneration by wind sid and star yet. And so, PR is ready for testing.

**Comment by 2hwk** [comment]
This needs another rebase to exp unfortunately

**Comment by 2hwk** [comment]
Is this dependant on https://github.com/flybywiresim/a32nx/pull/6528 ? Will have to wait for that to be merged to see the diff (or retarget flypadosv3 branch)

**Comment by tracernz** [comment]
Honestly, I don't think this can merge at all. Discontinuities should *not* be deleted, the ATSU should not be reaching into internal flight plan things, and the reason we don't import SID/STAR is because IRL ATSUs do not.

**Comment by igor8518** [comment]
> Honestly, I don't think this can merge at all. Discontinuities should _not_ be deleted, the ATSU should not be reaching into internal flight plan things, and the reason we don't import SID/STAR is because IRL ATSUs do not.

I do it more to help beginners. To switch their attention to more "interesting" moments.  You can consider it as a cheat. :-)

**Comment by 2hwk** [comment]
From what I gather from tracer it's more a case of its more that we have to support and maintain and we aren't prepared to do so right now.

Mainly because some of these aspects are going to be reworked. For example, passenger loading will move into the EFB after flypad OS 3 in the near term, we want to be able to save panel states, including the MCDU, as well as rework the MCDU to have 2 independent FMGC (and thus 2/split MCDU). 

So at best your changes would only last until 0.9.x most likely. 

In the future, we may provide a robust API so that if you wish to add onto the functionality of the project and help to automate functions (i.e. simulating the co-pilot), it may be possible to do so, but right now everything is still in very active development.

I am very much are interested in improving QOL (quality of life) for users, but we have to build on shifting sand and be realistic about what we can/can't support at this point in time.

**Comment by igor8518** [comment]
> From what I gather from tracer it's more a case of its more that we have to support and maintain and we aren't prepared to do so right now.
> 
> Mainly because some of these aspects are going to be reworked. For example, passenger loading will move into the EFB after flypad OS 3 in the near term, we want to be able to save panel states, including the MCDU, as well as rework the MCDU to have 2 independent FMGC (and thus 2/split MCDU).
> 
> So at best your changes would only last until 0.9.x most likely.
> 
> In the future, we may provide a robust API so that if you wish to add onto the functionality of the project and help to automate functions (i.e. simulating the co-pilot), it may be possible to do so, but right now everything is still in very active development.
> 
> I am very much are interested in improving QOL (quality of life) for users, but we have to build on shifting sand and be realistic about what we can/can't support at this point in time.

I heard you. I try to react in time to changes in a32nx. 
By the way, I am developing my own version of the api resembling PMDG.)

**Comment by igor8518** [comment]
With your permission, I will leave this pr open? I still invite you to test the logic.

**Comment by tracernz** [comment]
We are also in the process of replacing the entire flight plan/stringing system and nav database in https://github.com/flybywiresim/a32nx/tree/fms-v2

**Pull request closed without merging**